lld/ELF/Relocations.cpp

```diff
@@ -1659,10 +1659,16 @@ static bool handleNonPreemptibleIfunc(Symbol &sym, uint16_t flags) {
   // original section/value pairs. For non-GOT non-PLT relocation case below, we
   // may alter section/value, so create a copy of the symbol to make
   // section/value fixed.
+  //
+  // Prior to Android V, there was a bug that caused RELR relocations to be
+  // applied after packed relocations. This meant that resolvers referenced
+  // IRELATIVE relocations in the packed relocation section could not read
```

Comment:
Suggest more accurate wording: "would have been able to read globals with RELR relocations before they were relocated" (likewise below).

Reply:
Oh, I am confused by the sentence "... would have been able to read globals with RELR relocations before they were relocated".

Isn't it "could not ... " because RELR relocations have been applied yet?

Reply:
They can still read it because the memory is mapped as readable, it's just that the contents will be wrong.

Reply:
Switched to

> This meant that resolvers referenced
  // IRELATIVE relocations in the packed relocation section would read
  // unrelocated globals with RELR relocations when
  // --pack-relative-relocs=android+relr is enabled.

hope it is clearer

Reply:
"referenced" -> "referenced by"

Reply:
Fixed